static/app/api.tsx

```diff
@@ -445,71 +445,75 @@ export class Client {
+        let errorReason = 'Request not OK'; // the default error reason
+
+        // Try to get text out of the response no matter the status
         try {
-          responseJSON = await responseClone.json();
+          responseText = await response.text();
         } catch (error) {
+          ok = false;
           if (error.name === 'AbortError') {
-            ok = false;
             errorReason = 'Request was aborted';
-          } else if (isResponseJSON && error instanceof SyntaxError) {
-            // If the MIME type is `application/json` but decoding failed,
-            // this should be an error.
-            ok = false;
-            errorReason = 'JSON parse error';
+          } else {
+            errorReason = error.toString();
           }
         }
-      }
 
-      const responseMeta: ResponseMeta = {
-        status,
-        statusText,
-        responseJSON,
-        responseText,
-        getResponseHeader: (header: string) => response.headers.get(header),
-      };
-
-      // Respect the response content-type header
-      const responseData = isResponseJSON ? responseJSON : responseText;
-
-      if (ok) {
-        successHandler(responseMeta, statusText, responseData);
-      } else {
-        globalErrorHandlers.forEach(handler => handler(responseMeta));
-        errorHandler(responseMeta, statusText, errorReason);
-      }
+        const responseContentType = response.headers.get('content-type');
+        const isResponseJSON = responseContentType?.includes('json');
 
-      completeHandler(responseMeta, statusText);
-    });
+        const isStatus3XX = status >= 300 && status < 400;
+        if (status !== 204 && !isStatus3XX) {
+          try {
+            responseJSON = await responseClone.json();
+          } catch (error) {
+            if (error.name === 'AbortError') {
+              ok = false;
+              errorReason = 'Request was aborted';
+            } else if (isResponseJSON && error instanceof SyntaxError) {
+              // If the MIME type is `application/json` but decoding failed,
+              // this should be an error.
+              ok = false;
+              errorReason = 'JSON parse error';
+            }
+          }
+        }
+
+        const responseMeta: ResponseMeta = {
+          status,
+          statusText,
+          responseJSON,
+          responseText,
+          getResponseHeader: (header: string) => response.headers.get(header),
+        };
+
+        // Respect the response content-type header
+        const responseData = isResponseJSON ? responseJSON : responseText;
+
+        if (ok) {
+          successHandler(responseMeta, statusText, responseData);
+        } else {
+          globalErrorHandlers.forEach(handler => handler(responseMeta));
+          errorHandler(responseMeta, statusText, errorReason);
+        }
+
+        completeHandler(responseMeta, statusText);
+      })
+      .catch(() => {
+        // Ignore all failed requests
+      });
```

Comment:
Previously, [we were swallowing up these errors and sending them to Sentry](https://github.com/getsentry/sentry/pull/34653/files#diff-87c26d4c33b09c80ff6965eb0b192129cf0a53b73a4fe4b0893f6a67523c6303L553-L559), instead we will just ignore them.

Note this is different than the error handler if the fetch request succeeds, but we have a non 2xx response.